src/blk/kernel/KernelDevice.cc

```diff
@@ -99,16 +99,20 @@ int KernelDevice::choose_fd(FAMode buffered, int write_hint) const
 
 int KernelDevice::_lock()
 {
-  dout(10) << __func__ << " " << fd_directs[WRITE_LIFE_NOT_SET] << dendl;
+  dout(10) << __func__ << " " << choose_fd(FAMode::DIRECT) << dendl;
```

Comment:
> Fix locking if F_OFD_SETLK is not available

i am not sure how this change is related to locking? could you expand it a little bit in the commit message?

also, could you prefix the title of your commit message with the subcomponent your are changing ? see https://github.com/ceph/ceph/blob/master/SubmittingPatches.rst#3-describe-your-changes

Reply:
I could/should take it into a separate PR, since this PR is not about locking..
And the major part of the locking is that `F_OFD_SETLK` is a linux only define and locking method.
So for other systems we need to fall back to more basic locking.

Reply:
@tchaikov, 
Put the locking in a separate PR: #40864, so is this now commitable??